app/user/rest.py

```diff
@@ -558,15 +563,25 @@ def update_password(user_id):
     user = get_user_by_id(user_id=user_id)
     req_json = request.get_json()
     password = req_json.get("_password")
-
     user_update_password_schema_load_json.load(req_json)
+    add_password_for_user(user_id, password)
 
     current_app.logger.info("update_password", extra={"python_module": __name__, "user_id": user_id})
-
     update_user_password(user, password)
     return jsonify(data=user.serialize()), 200
 
 
+@user_blueprint.route("/<uuid:user_id>/check-password-validity", methods=["POST"])
```

Comment:
Are we adding to this function for the complexity check?  It should be a single call I feel